Generate serialization of CFDictionaryRef

achristensen07 opened the pull request:
#### 0d2bc1171d06de6b2816e04b2b292aaa00a68391
<pre>
Generate serialization of CFDictionaryRef
<a href="https://bugs.webkit.org/show_bug.cgi?id=268777">https://bugs.webkit.org/show_bug.cgi?id=268777</a>

Reviewed by Brady Eidson.

And clean up the other Array/Dictionary serialization.
Remove some unneeded allocations from unneeded UniqueRef pointers.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Cocoa/CoreIPCArray.h:
(WebKit::CoreIPCArray::CoreIPCArray):
* Source/WebKit/Shared/Cocoa/CoreIPCArray.mm:
(WebKit::CoreIPCArray::CoreIPCArray):
(WebKit::CoreIPCArray::toID const):
* Source/WebKit/Shared/Cocoa/CoreIPCArray.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCCFType.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.mm:
(WebKit::CoreIPCDictionary::CoreIPCDictionary):
(WebKit::CoreIPCDictionary::createNSDictionaryIfNeeded const):
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.h:
(WebKit::CoreIPCNSCFObject::value const):
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.serialization.in:
* Source/WebKit/Shared/cf/ArgumentCodersCF.cpp:
(IPC::ArgumentCoder&lt;CFDictionaryRef&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;RetainPtr&lt;CFDictionaryRef&gt;&gt;::decode): Deleted.
* Source/WebKit/Shared/cf/ArgumentCodersCF.h:
* Source/WebKit/Shared/cf/CFTypes.serialization.in:
* Source/WebKit/Shared/cf/CoreIPCCFArray.h:
(WebKit::CoreIPCCFArray::array const):
(WebKit::CoreIPCCFArray::CoreIPCCFArray): Deleted.
* Source/WebKit/Shared/cf/CoreIPCCFArray.mm:
(WebKit::CoreIPCCFArray::CoreIPCCFArray):
* Source/WebKit/Shared/cf/CoreIPCCFArray.serialization.in:
* Source/WebKit/Shared/cf/CoreIPCCFDictionary.h: Copied from Source/WebKit/Shared/cf/CoreIPCCFArray.h.
(WebKit::CoreIPCCFDictionary::vector const):
* Source/WebKit/Shared/cf/CoreIPCCFDictionary.mm: Copied from Source/WebKit/Shared/cf/CoreIPCCFArray.mm.
(WebKit::CoreIPCCFDictionary::CoreIPCCFDictionary):
(WebKit::CoreIPCCFDictionary::createCFDictionary const):
* Source/WebKit/Shared/cf/CoreIPCCFDictionary.serialization.in: Copied from Source/WebKit/Shared/cf/CoreIPCCFArray.serialization.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(operator==):
(dictionariesEqual):
(TEST):

Canonical link: <a href="https://commits.webkit.org/274192@main">https://commits.webkit.org/274192@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0ea0a6cd52744f7dd7abaf5f13da4d5f755ed845

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/38232 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17174 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/40545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/40790 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/34011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/40462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19927 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/14515 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/40790 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/38805 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/14495 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/40545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/40790 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/12590 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/40545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42067 "Failed to checkout and rebase branch from PR 23883") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/34751 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/40545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/42067 "Failed to checkout and rebase branch from PR 23883") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/13180 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/14515 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/42067 "Failed to checkout and rebase branch from PR 23883") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/14732 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/40545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/13595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4972 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/14196 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

beidson reviewed (approved):
Yay for unique ref removal

achristensen07 committed:
Generate serialization of CFDictionaryRef
https://bugs.webkit.org/show_bug.cgi?id=268777

Reviewed by Brady Eidson.

And clean up the other Array/Dictionary serialization.
Remove some unneeded allocations from unneeded UniqueRef pointers.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Cocoa/CoreIPCArray.h:
(WebKit::CoreIPCArray::CoreIPCArray):
* Source/WebKit/Shared/Cocoa/CoreIPCArray.mm:
(WebKit::CoreIPCArray::CoreIPCArray):
(WebKit::CoreIPCArray::toID const):
* Source/WebKit/Shared/Cocoa/CoreIPCArray.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCCFType.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.mm:
(WebKit::CoreIPCDictionary::CoreIPCDictionary):
(WebKit::CoreIPCDictionary::createNSDictionaryIfNeeded const):
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.h:
(WebKit::CoreIPCNSCFObject::value const):
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.serialization.in:
* Source/WebKit/Shared/cf/ArgumentCodersCF.cpp:
(IPC::ArgumentCoder<CFDictionaryRef>::encode): Deleted.
(IPC::ArgumentCoder<RetainPtr<CFDictionaryRef>>::decode): Deleted.
* Source/WebKit/Shared/cf/ArgumentCodersCF.h:
* Source/WebKit/Shared/cf/CFTypes.serialization.in:
* Source/WebKit/Shared/cf/CoreIPCCFArray.h:
(WebKit::CoreIPCCFArray::array const):
(WebKit::CoreIPCCFArray::CoreIPCCFArray): Deleted.
* Source/WebKit/Shared/cf/CoreIPCCFArray.mm:
(WebKit::CoreIPCCFArray::CoreIPCCFArray):
* Source/WebKit/Shared/cf/CoreIPCCFArray.serialization.in:
* Source/WebKit/Shared/cf/CoreIPCCFDictionary.h: Copied from Source/WebKit/Shared/cf/CoreIPCCFArray.h.
(WebKit::CoreIPCCFDictionary::vector const):
* Source/WebKit/Shared/cf/CoreIPCCFDictionary.mm: Copied from Source/WebKit/Shared/cf/CoreIPCCFArray.mm.
(WebKit::CoreIPCCFDictionary::CoreIPCCFDictionary):
(WebKit::CoreIPCCFDictionary::createCFDictionary const):
* Source/WebKit/Shared/cf/CoreIPCCFDictionary.serialization.in: Copied from Source/WebKit/Shared/cf/CoreIPCCFArray.serialization.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(operator==):
(dictionariesEqual):
(TEST):

Canonical link: https://commits.webkit.org/274192@main
